tests/storage/stop_status_utils.py

```
@@ -0,0 +1,32 @@
+        True if restarts exceed threshold.
+
+    Example::
+        dv.wait_for_dv_success(
+            stop_status_func=dv_stop_status_after_three_restarts,
+            dv=dv,
+            restart_count_threshold=4,
+        )
+    """
+    restart_count = getattr(dv.instance.status, "restartCount", 0) or 0
+    if restart_count >= restart_count_threshold:
+        LOGGER.error(f"DV {dv.name} has {restart_count} restarts, stopping")
```

Ruff (0.15.18)

[warning] Logging statement uses f-string

(G004)